cfme/tests/services/test_myservice.py

```diff
@@ -170,6 +172,39 @@ def test_service_link(appliance, context, myservice, provider):
         assert new_view.wait_displayed()
 
 
+@pytest.mark.parametrize('context', [ViaUI])
+@pytest.mark.meta(automates=[BZ(1720338)])
+def test_retire_service_with_retired_vm(appliance, context, myservice, provider):
+    """Tests retire service with an already retired vm.
+
+    Metadata:
+        test_flag: provision
+
+    Polarion:
+        assignee: nansari
+        casecomponent: Services
+        initialEstimate: 1/4h
+        tags: service
+
+    Bugzilla:
+        1720338
+    """
+    service_name, vm_name = myservice
+    vm = appliance.collections.infra_vms.instantiate(vm_name, provider)
+    vm.retire()
+    # using rest entity to check if the VM has retired since it is a lot faster
+    retire_vm = appliance.rest_api.collections.vms.get(name=vm_name)
```

Comment:
:+1: 
@sshveta optional or request. I found we need `rest_etntity` access for most of `BaseEntity` objects. let's add this as property in `BaseVM` class so you can easily access it with VM object. It will more useful for `rest` and `UI` access of a base entity objects.
```
    @property
    def rest_api_entity(self):
        return self.appliance.rest_api.collections.vms.get(name=self.name)
```

Reply:
Hi Nikhil , 

I will ask Parthvi (or FA owner ) to add it if that's okay .

Reply:
We can ultimately push this even farther to use sentaku to resolve the collection/entity `viaREST` or `viaUI`